staging/src/k8s.io/client-go/tools/clientcmd/config.go

```diff
@@ -58,6 +58,12 @@ type PathOptions struct {
 	LoadingRules *ClientConfigLoadingRules
 }
 
+var (
+	// UseConfigLock ensures that access to kubeconfig files is being guarded by a lock file
+	// Users can opt-out from using it by setting KUBECONFIG_NO_LOCK environment variable
+	UseConfigLock bool = len(os.Getenv("DISABLE_KUBECONFIG_LOCK")) == 0
```

Comment:
This should actually parse the bool for equivalence to true.  I'd like to prefix with a `KUBE_` on the env var.

You should also explain that this is public because it allows different binaries to directly change the default behavior.

Reply:
Since the only consumers of this envvar are kubectl user, I'm inclined to move that env bit entirely to kubectl.

Reply:
@soltysh this now doesn't match the PR description BTW

Reply:
You still have that ability, only not directly. Only consumers of the library have that ability.